applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -226,11 +319,7 @@ static void dvfs_retry_work_handler(struct k_work *work)
 {
 	LOG_DBG("Retrying to change DVFS frequency.");
 
-	enum dvfs_frequency_setting required_freq = check_required_frequency();
-
-	__ASSERT_NO_MSG(required_freq != current_freq);
-
-	set_dvfs_freq(required_freq);
+	set_dvfs_based_on_frequency();
```

Comment:
Expanding module state machine (as I suggested in one of my previous comments) would improve readability here and make the implementation less error prone (so that similar problems could be avoided).